packages/vjudge/src/model.ts

```diff
@@ -55,6 +55,11 @@ class Service {
 
     async sync(domainId: string, resync = false, list: string) {
         let page = 1;
+        if (domainId !== 'codeforces') {
```

Comment:
this **shouldn't** be hardcoded into model.
what if some other endpoints also add problem at the first page? (or middle?)
at current, if resyncing check for every page is required, emitting a problem list for **each** page works.
(although its not recommented, the update check runs every time after startup.)

Reply:
Maybe comparing problem count before syncing all page is a better solution.
Whether how adapter implements the optional `countProblem(list: string): Promise<number>` function depends on platform. Direct count API, count by page, fetch last page and get element count, etc.